From 72e04c644de36a1fd446aed3d8727fd5d7d127d6 Mon Sep 17 00:00:00 2001
From: Michel JAOUEN <michel.jaouen@st.com>
Date: Tue, 23 Sep 2025 11:40:30 +0200
Subject: [PATCH 931/954] fix(stm32mp2): increase scmi timeout

Since scmi request are now performed on non secure, delay need to be
increased especially when non secure test calling secure service are
played in non secure application

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git d04f991726eb770f2062e03216676decef84a9e3 fix(stm32mp2): increase scmi timeout]
Change-Id: I9352c5503cfed9ac52b488c08ae6f4425a6d0c09
Signed-off-by: Michel JAOUEN <michel.jaouen@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/502081
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Maxime MERE <maxime.mere@foss.st.com>
---
 plat/st/stm32mp2/scmi/scmi_common.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/plat/st/stm32mp2/scmi/scmi_common.c b/plat/st/stm32mp2/scmi/scmi_common.c
index 22f4f47c24..74cbab136b 100644
--- a/plat/st/stm32mp2/scmi/scmi_common.c
+++ b/plat/st/stm32mp2/scmi/scmi_common.c
@@ -12,7 +12,7 @@
 
 #include "scmi_private.h"
 
-#define TIMEOUT_10MS_IN_US		10000U
+#define TIMEOUT_IN_US			500000U
 
 /* IPCC chanel used for ringbell for SCMI mailbox */
 #define IPCC_CHANNEL15			14U
@@ -151,7 +151,7 @@ bool scmi_channel_error(void)
  */
 static int32_t scmi_channel_check(void)
 {
-	uint64_t timeout_ref = timeout_init_us(TIMEOUT_10MS_IN_US);
+	uint64_t timeout_ref = timeout_init_us(TIMEOUT_IN_US);
 
 	while (scmi_channel_busy()) {
 		if (scmi_channel_error() || timeout_elapsed(timeout_ref)) {
@@ -256,7 +256,7 @@ static int32_t scmi_proto_version(uint32_t proto_id, uint32_t *version)
 
 	scmi_ring_doorbell();
 
-	ret = scmi_rsp_wait(TIMEOUT_10MS_IN_US, 2U);
+	ret = scmi_rsp_wait(TIMEOUT_IN_US, 2U);
 	if (ret != SCMI_SUCCESS) {
 		return ret;
 	}
@@ -297,7 +297,7 @@ static int32_t scmi_proto_msg_attr(uint32_t proto_id, uint32_t command_id, uint3
 
 	scmi_ring_doorbell();
 
-	ret = scmi_rsp_wait(TIMEOUT_10MS_IN_US, 2U);
+	ret = scmi_rsp_wait(TIMEOUT_IN_US, 2U);
 	if (ret != SCMI_SUCCESS) {
 		return ret;
 	}
@@ -422,7 +422,7 @@ int32_t scmi_sys_pwr_state_set(uint32_t flags, uint32_t system_state)
 
 	scmi_ring_doorbell();
 
-	ret = scmi_rsp_wait(TIMEOUT_10MS_IN_US, 1U);
+	ret = scmi_rsp_wait(TIMEOUT_IN_US, 1U);
 	if (ret != SCMI_SUCCESS) {
 		return ret;
 	}
@@ -446,7 +446,7 @@ int32_t scmi_base_discover_list_proto(uint32_t *num, uint8_t *protocols, uint8_t
 
 	scmi_ring_doorbell();
 
-	ret = scmi_rsp_wait(TIMEOUT_10MS_IN_US, SIZE_NO_CHECK);
+	ret = scmi_rsp_wait(TIMEOUT_IN_US, SIZE_NO_CHECK);
 	if (ret != SCMI_SUCCESS) {
 		return ret;
 	}
-- 
2.53.0

